vortex-array/src/arrays/bool/stats.rs

```diff
@@ -251,7 +251,7 @@ mod test {
         let bool_arr = BoolArray::new(BooleanBuffer::new_set(0), Validity::NonNullable);
         assert!(bool_arr.statistics().compute_is_strict_sorted().is_none());
         assert!(bool_arr.statistics().compute_is_sorted().is_none());
-        assert!(bool_arr.statistics().compute_is_constant().is_none());
+        // assert!(bool_arr.statistics().compute_is_constant().is_none());
```

Comment:
@robert3005 do you remember the reason here? My intuition is that an empty array is always constant.

Reply:
Yes, we assume that for constant arrays we can get the single value out of it. The Array::as_constant breaks otherwise. You can always change the semantics

Reply:
I'll keep that for now, I've found a couple of codepaths that I think will break if I change that, and I can do that in a separate smaller PR.